src/main/java/net/minecraftforge/common/extensions/IForgeEntity.java

```diff
@@ -88,6 +89,9 @@ default ItemStack getPickedResult(HitResult target)
         ItemStack result = self().getPickResult();
         if (result == null) {
             SpawnEggItem egg = SpawnEggItem.byId(self().getType());
+            if (egg == null) {
```

Comment:
Honestly I would prefer this didn't exist at all, because modders *should* be implementing getPickResult themselves and this making this moot. However, once you change ForgeSpawnEggItem.get to call vanilla's then this can be changed to a single call.